compiler/luci/service/src/CircleShapeInferenceHelper.cpp

```diff
@@ -180,6 +180,11 @@ loco::TensorShape pad_shape(const loco::TensorShape &input_shape, const luci::Ci
   loco::TensorShape output_shape;
 
   output_shape.rank(input_shape.rank());
+
+  auto const_padding = dynamic_cast<const luci::CircleConst *>(paddings);
+  if (const_padding == nullptr)
+    return output_shape;
```

Comment:
> Just to clarify, does this mean that if the padding node is not constant, it does nothing, right?

Yes. when the `paddings` is not constant, it does nothing and return output shape filled with unknown dimensions.

Reply:
Just above this line, there is 
```
output_shape.rank(input_shape.rank());
```
which implicitly sets all dimes to unknown.
In another PR, add a comment about this would help others reading the code.